scripts/NFTVesting/test-deploy/linear/deploy_linear_time_lock_release_max.js

```diff
@@ -0,0 +1,170 @@
+
+	// Send NFT to timelock contract
+	await basicNFTInstance.transferFrom(
+		nftLocker.address,
+		timeLockInstance.address,
+		0 // token id
+	);
+
+	// check nftLocker of NFT after transfer to be timelock contract
+	const nftOwnerAfterTransfer = await basicNFTInstance.ownerOf(0);
+	console.log("nftOwnerAfterTransfer: ", nftOwnerAfterTransfer);
+
+	// Unhappy path to see if it can be released. Should not unlock.
+	// Set new timestamp by speeding up time
+	await ethers.provider.send("evm_setNextBlockTimestamp", [
+		timestampBefore + 50,
+	]);
+
+	// try release NFT
+	console.log("Releasing NFT prior to release time...");
+	try {
+		await timeLockInstance.release();
+	} catch (e) {
+		expect(e.reason).to.equal(
+			"Error: VM Exception while processing transaction: reverted with reason string 'BasicNFTTimelock: current time is before release time'"
+		);
+	}
+
+	// Set new timestamp by speeding up time
+	await ethers.provider.send("evm_setNextBlockTimestamp", [
+		timestampBefore + 999999,
+	]);
+	await ethers.provider.send("evm_mine"); // Fast forward time
+
+	// Get new timestamp
+	const blockNumAfter = await ethers.provider.getBlockNumber();
+	const blockAfter = await ethers.provider.getBlock(blockNumAfter);
+	const currentTimeStamp = blockAfter.timestamp;
+	console.log("currentTimeStamp: ", currentTimeStamp, "\n");
+
+	// Get balance of timelock contract
+	const timeLockBalance = await ethers.provider.getBalance(
+		timeLockInstance.address
+	);
+	console.log("timeLockBalance: ", ethers.utils.formatEther(timeLockBalance));
+
+	// release NFT
+	console.log(
+		"Release Time: ",
+		ethers.utils.formatUnits(await timeLockInstance.vestingStartTime(), 18 - 18)
+	);
+
+	console.log(
+		"Release Time: ",
+		ethers.utils.formatUnits(await timeLockInstance.vestingStartTime(), 18 - 18)
+	);
+
+	const blockNumBeforeRelease = await ethers.provider.getBlockNumber();
+	const blockBeforeRelease = await ethers.provider.getBlock(
+		blockNumBeforeRelease
+	);
+	console.log("Time Before Release:", blockBeforeRelease.timestamp, "\n");
+
+	// Get current discount
+	const currentDiscount2 = await timeLockInstance.getDiscount();
+	expect(currentDiscount2).to.deep.equal(ethers.utils.parseEther("1000"));
+
+	console.log("Releasing NFT... \n");
+
+	const releaseTx = await timeLockInstance.release();
+	const newNftOwner = await basicNFTInstance.ownerOf(0);
+	console.log("newNftOwner: ", newNftOwner); // same as original beneficiary
+
+	// Get the ETH balance of NFT Locker and Beneficiary
+	const nftLockerBalanceAfter = await ethers.provider.getBalance(
+		nftLocker.address
+	);
+	const beneficiaryBalanceAfter = await ethers.provider.getBalance(
+		beneficiary.address
+	);
+
+	console.log(
+		"nftLockerBalanceAfter: ",
+		ethers.utils.formatEther(nftLockerBalanceAfter)
+	);
+	console.log(
+		"beneficiaryBalanceAfter: ",
+		ethers.utils.formatEther(beneficiaryBalanceAfter)
+	);
+
+	// // check if {release} function can be called again. Expected failure.
```

Comment:
Same

Reply:
Seems like it's repeated, maybe we need to test this from another script or just remove

Reply:
dont think its needed